### ISSUE TITLE
Rollback option on workload does not work

### ISSUE THREAD

**Issue by sowmyav27** [rancher/dashboard#4664]
**Rancher Server Setup**
- Rancher version: 2.6-head and 2.6.2
- Installation option (Docker install/Helm Chart): docker install

**Information about the Cluster**
- Kubernetes version: 1.21
- Cluster Type (Local/Downstream): EKS Downstream

**Describe the bug**
Rollback on a workload doesnt work

**To Reproduce**

- Create a deployments with image1
- Edit and change image to image2
- On the workload, select `Rollback` and choose revision1
- Error seen `deployments.apps "test" not found`

```
apiVersion: "v1"
code: 404
details: {name: "test", group: "apps", kind: "deployments"}
kind: "Status"
message: "deployments.apps \"test\" not found"
metadata: {}
reason: "NotFound"
status: "Failure"
```

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by pacholoamit** [comment]
Yeah, noticing this as well. Screwed up my deployment a few minutes ago. Glad it wasn't as critical but the rollback feature doesn't work.

**Comment by sgapanovich** [comment]
While testing https://github.com/rancher/dashboard/issues/4824 noticed this issue as well. Investigated it and wanted to open an issue but I see we already created one 👍🏻 

Going to share my debugging here. 
Rolling back on UI in a local cluster works as expected.
The issue happens in a downstream cluster because of the reason explained by @ryansann 

I was able to manually to rollback using a PATCH request.
First I rolled-back using kubectl and got the whole path used by kubectl
![rollback v6](https://user-images.githubusercontent.com/40282141/151484335-634f056a-9f0a-4c8b-835f-04ba1f25ef7f.jpg)

I copied the request body which is sent by UI and sent the request manually to the url used by kubectl 
`PATCH https://<rancher-url>/k8s/clusters/<cluster_id>/apis/apps/v1/namespaces/default/deployments/testing` and was able to rollback the deployment 

note: `Content-Type` in the request header should be `application/json-patch+json`

![payload](https://user-images.githubusercontent.com/40282141/151485289-7f086dad-9242-42d9-b512-d3ac264a2374.jpg)

**Comment by gaktive** [comment]
Internal reference: SURE-3964
Also plenty of feedback about its brokenness here: https://github.com/rancher/rancher/issues/35138 (which is earlier than this ticket)

**Comment by ronhorton** [comment]
**Pass** Verified fixed in `2.6-head Commit ID f2a4147`

1. created deployment with nginx 1.21.3
2. edit config on deployment to nginx:latest
3. rollback to previous deployment (1.21.3)
4. show diff 
5. rollback

`rollback to previous version functions as expected`

**Comment by SGudbrandsson** [comment]
Does this also fix the rollbacks not being in a cronological order?
